### PR TITLE
[AF-681] Add SUNSETTING and SUNSETTED

### PR DESCRIPTION
**Pull request by nick-f**
:v:

/cc @zendesk/vegemite

### Description
Add support for arrays of versions as well as `sunsetting` and `sunsetted` versions.

This can be used to display warnings in ZAT, such as when running [`zat update`](https://github.com/zendesk/zendesk_apps_tools/pull/190).

### References
* JIRA: https://zendesk.atlassian.net/browse/AF-681

### Risks
* [low] New methods being added

### CODE REVIEW

File: lib/zendesk_apps_support/app_version.rb [hunk 1]
```diff
@@ -31,7 +35,15 @@ def valid_for_update?
     end
 
     def deprecated?
-      @version == DEPRECATED
+      DEPRECATED.include?(@version)
+    end
+
+    def sunsetting?
+      SUNSETTING && SUNSETTING.include?(@version)
```
Comment (ocke) [inline lib/zendesk_apps_support/app_version.rb]: you forgot the leading `true &&`

Comment (ocke) [inline lib/zendesk_apps_support/app_version.rb]: Isn't `SUNSETTING` always true?

File: lib/zendesk_apps_support/app_version.rb [hunk 2]
```diff
@@ -31,7 +35,15 @@ def valid_for_update?
...
+    end
+
+    def current
```
Comment (ocke) [inline lib/zendesk_apps_support/app_version.rb]: `attr_reader` ? @danielbreves

File: lib/zendesk_apps_support/app_version.rb (hunk 1 above)
Comment (nick-f) [inline lib/zendesk_apps_support/app_version.rb]: At the moment, yes, but this is preparing for when there are no versions currently being sunset which will happen sometime in the future.

File: lib/zendesk_apps_support/app_version.rb (hunk 2 above)
Comment (nick-f) [inline lib/zendesk_apps_support/app_version.rb]: 💇‍♂️

File: lib/zendesk_apps_support/app_version.rb [hunk 3]
```diff
@@ -1,24 +1,31 @@
...
-    TO_BE_SERVED     = [DEPRECATED, CURRENT, FUTURE].compact.freeze
-    VALID_FOR_UPDATE = [CURRENT, FUTURE].compact.freeze
+    TO_BE_SERVED     = [DEPRECATED, SUNSETTING, CURRENT, FUTURE].compact.flatten.freeze
+    VALID_FOR_UPDATE = [SUNSETTING, CURRENT, FUTURE].compact.flatten.freeze
+
+    attr_reader :current
 
     def initialize(version)
       @version = version.to_s
       @version.freeze
```
Comment (fvzzy) [inline lib/zendesk_apps_support/app_version.rb]: nitpick: could be chained (?)

File: lib/zendesk_apps_support/validations/manifest.rb [hunk 4]
```diff
@@ -292,7 +292,7 @@ def missing_framework_version(manifest)
         end
 
         def invalid_version_error(manifest, package)
-          valid_to_serve = AppVersion::TO_BE_SERVED
+          valid_to_serve = AppVersion::TO_BE_SERVED.compact
```
Comment (fvzzy) [inline lib/zendesk_apps_support/validations/manifest.rb]: Isn't this const already `compact`ed in the class?

File: lib/zendesk_apps_support/app_version.rb [hunk 5]
```diff
@@ -31,7 +38,11 @@ def valid_for_update?
     end
 
     def deprecated?
-      @version == DEPRECATED
+      DEPRECATED.include?(@version)
```
Comment (danielbreves) [inline lib/zendesk_apps_support/app_version.rb]: Why did you need this change?

File: lib/zendesk_apps_support/validations/manifest.rb (hunk 4 above)
Comment (danielbreves) [inline lib/zendesk_apps_support/validations/manifest.rb]: Isn't `TO_BE_SERVED` already `compact`?

Comment (nick-f) [inline lib/zendesk_apps_support/validations/manifest.rb]: Good pickup. Fixed.

File: lib/zendesk_apps_support/app_version.rb (hunk 5 above)
Comment (nick-f) [inline lib/zendesk_apps_support/app_version.rb]: Allows for `DEPRECATED` to be an array of values. Or, when 1.0 is deprecated, will we just drop 0.5 from the list entirely? If the latter is true, then this can be removed.

File: lib/zendesk_apps_support/validations/manifest.rb (hunk 4 above)
Comment (nick-f) [inline lib/zendesk_apps_support/validations/manifest.rb]: Indeed it is, good pickup. Fixed.

File: lib/zendesk_apps_support/app_version.rb (hunk 5 above)
Comment (danielbreves) [inline lib/zendesk_apps_support/app_version.rb]: Considering this is a public constant I think it would be dangerous to change its type. See https://github.com/zendesk/zendesk_app_market/search?utf8=%E2%9C%93&q=AppVersion&type=.

Comment (nick-f) [inline lib/zendesk_apps_support/app_version.rb]: Ah, I see. Reverted back to how it was before. 💇‍♂️
